### PR TITLE
teams: testing hidden key rotations

### PR DESCRIPTION
**Pull request by maxtaco**
- make a successful test run for the loader Unit tests with hidden rotations
- fix a few small bugs where the client and server disagreed
- fix a real bug where the client wasn't properly computing the max ratchet
- add some machinery for skipping units
- first batch, more on the way (and likely needs more tweaks)

### CODE REVIEW

**Comment by maxtaco** [comment]
(See https://github.com/keybase/keybase-test-vectors/pull/23 for the corresponding new vectors)

**Comment by maxtaco** [comment]
This likely needs the news tests merged and vendored before it can pass cI

File: go/teams/keys.go [hunk 1]
```diff
@@ -433,7 +433,8 @@ func computeSeedCheck(id keybase1.TeamID, seed keybase1.PerTeamKeySeed, prev *ke
 	g := func(seed keybase1.PerTeamKeySeed, prev keybase1.PerTeamSeedCheckValue) keybase1.PerTeamSeedCheckValue {
 		digest := hmac.New(sha512.New, seed[:])
 		digest.Write([]byte(prev))
-		return keybase1.PerTeamSeedCheckValue(digest.Sum(nil)[:32])
+		sum := digest.Sum(nil)[:32]
+		return keybase1.PerTeamSeedCheckValue(sum)
```
Comment (heronhaye) [inline go/teams/keys.go]: is there any logical change here, or just style?

Comment (maxtaco) [reply go/teams/keys.go]: hangover from debug dumping, i just didn't bother to change it back

**Comment by maxtaco** [comment]
CI failure is in master too, going with a merge!

**Pull request merged**